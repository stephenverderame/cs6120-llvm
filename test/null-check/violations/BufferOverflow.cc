#include <cassert>
#include <cstdio>
#include <cstdlib>
// NOLINTBEGIN
int main(int argc, const char** argv)
{
    if (argc > 256) {
        printf("Too many arguments\n");
        return 1;
    }
    char raw_buffer[100];
    assert(argc < 100);
    for (int i = 0; i < 100; ++i) {
        const auto t = argv + i;
        assert(t != nullptr);
        const auto word = *t;
        assert(word != nullptr);
        int num = atoi(word);
        *(int*)(&raw_buffer[i]) = num;
    }
}
// NOLINTEND