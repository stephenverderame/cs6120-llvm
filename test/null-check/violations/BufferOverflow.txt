[31mSafety Violation[00m: Use of pointer that may overflow in store at BufferOverflow.cc:19
Because an index could not be proven to be in range, the following counterexample was found:
	[1mi[0m = 98
