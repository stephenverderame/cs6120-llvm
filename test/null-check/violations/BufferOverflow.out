[31mSafety Violation[00m: Use of pointer that may overflow in store at BufferOverflow.cc:19
